### PR TITLE
Refactor search  page

### PR DESCRIPTION
**Pull request by grafixeyehero**
**Changes**
- Replace Legacy `Api` to new `SDK API`

### CODE REVIEW

File: src/components/cardbuilder/Card/useCard.ts [hunk 1]
```diff
@@ -81,6 +82,8 @@ function useCard({ item, cardOptions }: UseCardProps) {
         cardOptions.cardCssClass,
         cardOptions.cardClass,
         { 'card-hoverable': layoutManager.desktop },
+        { 'show-focus': layoutManager.tv },
+        { 'show-animation': layoutManager.tv && enableFocusTransform },
```
Comment (dmitrylyzo) [inline src/components/cardbuilder/Card/useCard.ts:86]: If focus indication is broken in 10.9, this should be extracted as bugfix.

Comment (grafixeyehero) [reply src/components/cardbuilder/Card/useCard.ts:86]: This is the first time i used the card in stable router.

Comment (grafixeyehero) [reply src/components/cardbuilder/Card/useCard.ts:86]: Fix Search Field for Tv Mode #5610

**Comment by thornbill** [comment]
We haven't used any mui components in tv mode yet... do they work as expected?

**Comment by grafixeyehero** [comment]
didn't test in actual tv.


https://github.com/jellyfin/jellyfin-web/assets/32230989/214dca79-2ea8-4dd2-9368-116bfd4f6282

**Comment by grafixeyehero** [comment]
@thornbill the last push was not include mui components change.

File: src/strings/en-us.json [hunk 2]
```diff
@@ -359,7 +360,6 @@
     "HeaderApiKeysHelp": "External applications are required to have an API key in order to communicate with the server. Keys are issued by logging in with a normal user account or manually granting the application a key.",
     "HeaderApp": "App",
     "HeaderAppearsOn": "Appears On",
-    "HeaderAudioBooks": "Audio Books",
```
Comment (thornbill) [inline src/strings/en-us.json:362]: Can we avoid these translation key changes? Translators would need to re-translate them for all languages. :sweat_smile:

File: src/hooks/searchHook/useSearchItems.ts [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+import { MediaType } from '@jellyfin/sdk/lib/generated-client/models/media-type';
+import { getItemsApi } from '@jellyfin/sdk/lib/utils/api/items-api';
+import { getPersonsApi } from '@jellyfin/sdk/lib/utils/api/persons-api';
+import { getArtistsApi } from '@jellyfin/sdk/lib/utils/api/artists-api';
+import { useQuery } from '@tanstack/react-query';
+import { useApi } from '../useApi';
+import type { CardOptions } from 'types/cardOptions';
+import { CardShape } from 'utils/card';
+
+const Parameters_OPTIONS = {
```
Comment (thornbill) [inline src/hooks/searchHook/useSearchItems.ts:21]:
```suggestion
const QUERY_OPTIONS = {
```

File: src/hooks/searchHook/useSearchItems.ts [hunk 4]
```diff
@@ -0,0 +1,509 @@
...
+const isMusic = (collectionType: string) =>
+    collectionType === CollectionType.Music;
+
+const isTVShows = (collectionType: string) =>
+    collectionType === CollectionType.Tvshows;
+
+const isLivetv = (collectionType: string) =>
+    collectionType === CollectionType.Livetv;
+
+const Livetv_CARD_OPTIONS = {
```
Comment (thornbill) [inline src/hooks/searchHook/useSearchItems.ts:96]:
```suggestion
const LIVETV_CARD_OPTIONS = {
```

File: src/components/search/SearchFields.tsx [hunk 5]
```diff
@@ -1,33 +1,31 @@
...
     const inputRef = useRef<HTMLInputElement>(null);
 
     const onAlphaPicked = useCallback((e: Event) => {
         const value = (e as CustomEvent).detail.value;
         const inputValue = inputRef.current?.value || '';
 
         if (value === 'backspace') {
-            onSearch(inputValue.length ? inputValue.substring(0, inputValue.length - 1) : '');
+            onSearch(inputValue.length ? inputValue.substring(0, inputValue.length - 1) : ''
+            );
```
Comment (thornbill) [inline src/components/search/SearchFields.tsx:28]:
```suggestion
            onSearch(inputValue.length ? inputValue.substring(0, inputValue.length - 1) : '');
```

**Review by thornbill — CHANGES_REQUESTED** [review]

**Review by thornbill — APPROVED** [review]

**Pull request merged**